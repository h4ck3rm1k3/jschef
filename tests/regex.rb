

VALID_CHAR = [
      0x9, 0xA, 0xD,
      (0x20..0xD7FF),
      (0xE000..0xFFFD),
      (0x10000..0x10FFFF)
]

VALID_XML_CHARS = Regexp.new('^['+
                             VALID_CHAR.map { |item|
                               case item
                               when Fixnum
                                 [item].pack('U').force_encoding('utf-8')
                               when Range
                                 [item.first, '-'.ord, item.last].pack('UUU').force_encoding('utf-8')
                               end
                             }.join +
                             ']*$')
