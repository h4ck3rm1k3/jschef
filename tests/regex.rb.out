+ ruby -I ../opal/lib/ ../opal/bin/opal tests/regex.rb

/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:4679
      throw exception;
      ^
pack: undefined method `pack' for [9]:Array
    at Opal.defs.TMP_1 [as $new] (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:4894:15)
    at Array.ːmethod_missing (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:3448:70)
    at Array.method_missing_stub [as $pack] (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:1079:35)
    at /tmp/opal-nodejs-runner-20160708-8803-1yd75h7:19805:57
    at Opal.cdecl.$scope.get.$new.$a.$$p.TMP_1 (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:19805:247)
    at Opal.yield1 (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:1271:14)
    at Array.ːcollect (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:11003:26)
    at /tmp/opal-nodejs-runner-20160708-8803-1yd75h7:19805:301
    at Object.<anonymous> (/tmp/opal-nodejs-runner-20160708-8803-1yd75h7:19806:3)
    at Module._compile (module.js:409:26)
